core/domain/user_jobs_one_off_test.py

```diff
@@ -196,6 +196,130 @@ def test_user_contributions_get_created_after_running_the_job(self):
+            id=self.USER_B_ID,
+            gae_id=self.USER_B_GAE_ID,
+            email=self.USER_B_EMAIL
+        )
+
+
+    def test_prior_to_one_off_job_user_auth_model_does_not_exist(self):
+        self.assertIsNone(user_models.UserAuthModel.get_by_id(self.USER_A_ID))
+        self.assertIsNone(user_models.UserAuthModel.get_by_id(self.USER_B_ID))
+        self.assertIsNone(
+            user_models.UserAuthModel.get_by_id(self.PROFILE_A_ID))
+
+    def test_one_off_job_for_user_with_no_additional_profile(self):
+        self.user_b_model.put()
+        self._run_one_off_job()
+        user_auth_models = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, self.USER_B_GAE_ID)
+        self.assertEqual(len(user_auth_models), 1)
+        self.assertEqual(user_auth_models[0].gae_id, self.USER_B_GAE_ID)
+        self.assertEqual(user_auth_models[0].id, self.USER_B_ID)
+
+    def test_one_off_job_for_user_with_additional_profile(self):
+        self.user_a_model.put()
+        self.profile_a_model.put()
+        self._run_one_off_job()
+        user_auth_models = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, self.USER_A_GAE_ID)
+        self.assertEqual(len(user_auth_models), 2)
+        user_id_auth_id_pairs_list = [
+            [item.id, item.gae_id] for item in user_auth_models
+        ]
+        self.assertIn(
+            [self.user_a_model.id, self.user_a_model.gae_id],
+            user_id_auth_id_pairs_list)
+        self.assertIn(
+            [self.profile_a_model.id, self.profile_a_model.gae_id],
+            user_id_auth_id_pairs_list)
+
+    def test_one_off_job_run_with_all_kinds_of_users_works_fine(self):
+        self.user_a_model.put()
+        self.user_b_model.put()
+        self.profile_a_model.put()
+        self._run_one_off_job()
+        user_auth_models = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, self.USER_B_GAE_ID)
+        self.assertEqual(len(user_auth_models), 1)
+        self.assertEqual(user_auth_models[0].gae_id, self.USER_B_GAE_ID)
+        self.assertEqual(user_auth_models[0].id, self.USER_B_ID)
+
+        user_auth_models = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, self.USER_A_GAE_ID)
+        self.assertEqual(len(user_auth_models), 2)
+        user_auth_id_pairs_list = [
+            [item.id, item.gae_id] for item in user_auth_models
+            ]
+        self.assertIn(
+            [self.user_a_model.id, self.user_a_model.gae_id],
+            user_auth_id_pairs_list)
+        self.assertIn(
+            [self.profile_a_model.id, self.profile_a_model.gae_id],
+            user_auth_id_pairs_list)
+
+    def test_one_off_job_gae_id_registered_for_another_user_raises_error(self):
+        self.user_a_model.put()
+        self._run_one_off_job()
+        self.user_b_model.gae_id = self.user_a_model.gae_id
+        self.user_b_model.put()
+        with self.assertRaisesRegexp(
```

Comment:
@ everyone!
Need help in the failing backend tests:
- `test_one_off_job_gae_id_registered_for_another_user_raises_error` and 
- `test_one_off_job_profile_registered_with_another_gae_id_raises_error`,

I am unable to understand [error message](https://github.com/oppia/oppia/pull/10123/checks#step:7:551) in the failing test. I have a `assertRaisesRegexp` condition to check whether the exception was raised or not. It seems that the exception isn't even being raised due to the error if I remove the `assertRaisesRegexp` in the code. Locally also I get a similar error like [this](https://github.com/oppia/oppia/pull/10123/checks#step:7:526) - `WARNING:root:Slice 1558342726402testbed-request-id-hash-0 0 failed for the 1 of up to 11 attempts (1 of 31 taskqueue execution attempts). Will retry now.`
I am unable to find the bug. If anyone can help, it would be great. Thanks!

Reply:
Tasks are run via a task queue. If a single task fails, it will print an error, not bubble the exception up to the test. See how other tests analyze the output of a test to verify if it failed correctly.

Note that you should probably be yielding an error in your job, not raising an exception. A raised exception means the task failed unexpectedly, so the task queue will try to restart the task. What you want to do is output an expected failure--the task succeeded with a detected failure case.

Reply:
Thanks @BenHenning for the help! I have used the class functionalities to get the error now. And the tests passed locally.